internal/pkg/template/diff/override.go

```diff
@@ -40,58 +43,185 @@ func (m *ignorer) parse(_, _ *yaml.Node, _ string, _ overrider) (diffNode, error
-	if from.Kind == yaml.MappingNode {
-		fullFormNode, shortFormNode = from, to
-	} else {
-		fullFormNode, shortFormNode = to, from
+	if _, ok := intrinsicFunctions[toFunc]; !ok {
+		return false
 	}
-	if len(fullFormNode.Content) != 2 {
-		// The full form mapping node always contain only one child node, whose key is the func name in full form.
-		// Read https://www.efekarakus.com/2020/05/30/deep-dive-go-yaml-cfn.html.
+	return fromFunc == toFunc
+}
+
+// intrinsicFuncMatcher matches and parses two intrinsic function nodes written in different form (full/short).
+type intrinsicFuncMapTagConverter struct {
+	intrinsicFunc intrinsicFuncMatcher
+}
+
+// match returns true if from and to node represent the same intrinsic function written in different (full/short) form.
+// Example1: "!Ref" and "Ref:" will return true.
+// Example2: "!Ref" and "!Ref" will return false because they are written in the same form (i.e. short).
+// Example3: "!Ref" and "Fn::GetAtt:" will return false because they are different intrinsic functions.
+// For more on intrinsic functions and full/short forms, read https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ToJsonString.html.
+func (converter *intrinsicFuncMapTagConverter) match(from, to *yaml.Node, key string, overrider overrider) bool {
+	if !converter.intrinsicFunc.match(from, to, key, overrider) {
 		return false
 	}
-	return eqIntrinsicFunc(intrinsicFuncFullFormName(fullFormNode), shortFormNode.Tag)
+	// Exactly one of from and to is full form.
+	return (from.Kind == yaml.MappingNode || to.Kind == yaml.MappingNode) && (from.Kind != to.Kind)
 }
 
-func (_ *intrinsicFuncFullShortFormConverter) parse(from, to *yaml.Node, key string, overrider overrider) (diffNode, error) {
+// parse compares two intrinsic function nodes written in different form (full vs. short).
+// When the inputs to the intrinsic functions have different data types, parse assumes that no type conversion is needed
+// for correct comparison.
+// E.g. given "!Func: [1,2]" and "Fn::Func: '1,2'", parse assumes that comparing [1,2] with "1,2" produces the desired result.
+// Note that this does not hold for "GetAtt" function: "!GetAtt: [1,2]" and "!GetAtt: 1.2" should be considered the same.
+// parse assumes that from and to are matched by intrinsicFuncMapTagConverter.
+func (*intrinsicFuncMapTagConverter) parse(from, to *yaml.Node, key string, overrider overrider) (diffNode, error) {
 	var diff diffNode
 	var err error
 	if from.Kind == yaml.MappingNode {
 		// The full form mapping node always contain only one child node. The second element in `Content` is the 
 		// value of the child node. Read https://www.efekarakus.com/2020/05/30/deep-dive-go-yaml-cfn.html.
-		diff, err = parse(from.Content[1], stripTag(to), intrinsicFuncFullFormName(from), overrider)
+		diff, err = parse(from.Content[1], stripTag(to), intrinsicFunctions[intrinsicFuncName(from)].fullName, overrider)
```

Comment:
would this panic in the case
```
SomeValue
  Fn::Sub:
```

Reply:
Tested - not panicking!

Are you concerned of `from.Content[1]`? When the yaml looks like your example, `from`'s `Content` would still have a length of two, and its second element (i.e. the value to the key `Fn::Sub`) would be `&{8 0 !!null   <nil> []    <line> <col>}`

Reply:
ok, awesome! yeah, that was the case i was thinking about. sounds good 👍